pkg/chartverifier/reportBuilder.go

```diff
@@ -150,3 +162,43 @@ func GenerateSha(rawFiles []*helmchart.File) string {
 
 	return fmt.Sprintf("sha256:%x", chartSha.Sum(nil))
 }
+
+func GetPackageDigest(uri string) string {
+
+	url, err := url.Parse(uri)
+	if err != nil {
+		return ""
+	}
+	var chartReader io.Reader
+	switch url.Scheme {
+	case "http", "https":
+		var chartGetResponse *http.Response
+		chartGetResponse, err = http.Get(url.String())
+		if err == nil {
+			chartReader = chartGetResponse.Body
+		}
+	case "file", "":
+		if strings.HasSuffix(url.Path, ".tgz") {
+			chartReader, _ = os.Open(url.Path)
+		}
+	default:
+		err = errors.Errorf("scheme %q not supported", url.Scheme)
```

Comment:
Can you think of an example for this non supported scheme?

Reply:
I took the lead here from https://github.com/redhat-certification/chart-verifier/blob/a857b7d4e6a393f8ac09339eaa9e31fc773cd6e8/pkg/chartverifier/checks/helm.go#L161

I guess in theory the user could enter anything although we probably would then fail way before we go to this code. Also difficult to write code which doesn't cover unexpected values.

Reply:
Just curious if you though of scenarios like that, but yeah is better to be save and check for unexpected values... defensive is good here.